Add Aeris theme

Giyuu12 opened the pull request:
I made a flexible theme for the Windows 11 Taskbar Styler mod. 
Could you please consider add this theme into the mod?

However, during the making of this theme, a Windows Update was released that changed the XAML structure related to search, which made it prone to crashing when modifying the search button's style. As a result, styling of the search button has been intentionally left out.

bbmaster123 commented:
looks very clean and nice, I will test it out as soon as I can :)

btw, the issue with the search pill, if its the one I believe you're referring to, is that setting a background causes it to crash explorer when pointer over or pressed, and has existed for a while. Is that the same bug? The one I'm talking about is only for the search pill, the smaller rounder one that turns blue when in the pressed state.
Leaving it not styled is acceptable!

Are you positive the xaml structure has been updated? May I ask what build you're on now? It does update from time to time but the last time I looked personally was a few months ago
cheers

Giyuu12 commented:
Thanks for saying it's nice!

I'm aware of the crash caused by styling the background of the search pill, but this is a different issue. Just switching the search button display mode from Windows settings can cause a crash—even when no styles are applied to the search button at all. So, it might be necessary to update the mod itself.

As for the XAML structure, it's like what you can see in the image. If I remember correctly, the search button's structure starts with something like Taskbar.TaskbarSearchButton. I'm on build 24H2, 26100.4351.
<img width="575" alt="explorer_1VCWGYVoLE" src="https://github.com/user-attachments/assets/8c082f12-6c5c-4578-b590-2b1a66d1c325" />

m417z commented:
> Just switching the search button display mode from Windows settings can cause a crash—even when no styles are applied to the search button at all. So, it might be necessary to update the mod itself.

Does the crash happen even if there are absolutely no styles applied? If not, can you try to find the style that causes the crash?

If only having the mod enabled, even with no styles at all, causes the crash, can you also check whether the crash happens with the mod disabled and with UWPSpy? If it happens with UWPSpy too then it might be an issue with Microsoft's XAML Diagnostics API.

Giyuu12 commented:
After setting Taskbar Styler to {}, selecting SearchUx.SearchUI.SearchButtonRootGrid - SearchBoxButtonRootPanel in UWPSpy, then changing the search button visibility from Windows Settings, and selecting the same element again, explorer.exe crashes.
This issue also occurs even when Taskbar Styler and Windhawk are disabled, and also while using the current Aeris theme.

Since I was always working with UWPSpy running, I didn't notice it before, but it seems that Taskbar Styler is not the cause.

bbmaster123 commented:
ugh not again! Well hopefully the xaml structure doesn't change too much...I still have the old structure for now.


With the current code, search has a transparent background due to the earlier target `Taskbar.TaskListLabeledButtonPanel@CommonStates > Border#BackgroundElement`

could you try adding:
```
Taskbar.SearchBoxButton > Taskbar.TaskListButtonPanel > Border#BackgroundElement 
Background={ThemeResource SearchPillButtonBackground}
```

and let me know if either, or both search styles have their default background?

aside from search, I was thinking the show desktop button could be made larger and squared like the rest of the buttons so they all match:

```
Rectangle#ShowDesktopPipe
Height=48
Width=14
Fill=$fillColor
```

Everything else looks good!

Giyuu12 commented:
> Can you please attach a similar screenshot to mine, with the full tree starting from the root, and the element that causes the crash marked in the screenshot?

this is my structure
![explorer_QN2zU8YZf9](https://github.com/user-attachments/assets/486ea4aa-6307-478d-af9a-623dd3b1f03a)

> With the current code, search has a transparent background due to the earlier target

I added a style to set the default background, but I can't test it in my environment. Could you please check if it works?
As for the ShowDesktop button, I changed the style to make it more consistent.

m417z commented:
Thanks, I updated the workaround, please confirm it by trying this version:
https://github.com/m417z/my-windhawk-mods/blob/f84e3915f2d634a3ed58a3619d55c0e749758745/mods/windows-11-taskbar-styler.wh.cpp

To try it, disable the existing mod, then click the bottom right button in Windhawk to create a new mod, replace the code with the test mod code, click "Compile Mod", then "Exit Editing Mode".

Giyuu12 commented:
> To try it, disable the existing mod, then click the bottom right button in Windhawk to create a new mod, replace the code with the test mod code, click "Compile Mod", then "Exit Editing Mode".



Thanks! The crash doesn't happen anymore.
But when I compared your code with the current version, it looks like your version can't style the search element using visual style anymore (it worked before).

example
```
Target:
SearchUx.SearchUI.SearchButtonRootGrid@CommonStates

Styles:
Background@ActiveNormal=#FF0000
```

I tried to figure out why, but I think it's better to leave it to you.

For now, I applied minimal styling to the search-related element without using visual style.

Giyuu12 commented:
> What exactly worked, and how did you come to the conclusion that it has the CommonStates visual state if clicking on the element with UWPSpy causes a crash?

In my environment, I’ve been able to access SearchUx.SearchUI.SearchButtonRootGrid and Taskbar.SearchBoxButton > Taskbar.TaskListButtonPanel under certain conditions — specifically when using only the search icon, not the search pill and full search bar.

![explorer_GwR4aazKdk](https://github.com/user-attachments/assets/aeb7fdee-98d5-42bb-9745-963ca0f1b7b4)

If I change the setting in Windows to use the search pill or search box, then trying to access these elements causes a crash.

When I said it worked before, I meant that in the non-crashing state (with the search icon), I was able to apply visual style and confirm that it changed things like colors.

However, using visual style also caused worse issues, like crashes when switching to the search pill from the settings, and UWPSpy crashes as well. So I think your solution to disable visual style is a good approach.

Giyuu12 commented:
×visual style
○visual states

Giyuu12 commented:
After reviewing the new source code, I saw that the visualstate is now ignored when the parent of SearchUx.SearchUI.SearchButtonRootGrid is SearchUx.SearchUI.SearchPillButton. When the search pill is not being used, the parent of SearchButtonRootGrid becomes SearchUx.SearchUI.SearchIconButton (the SearchPillButton hierarchy disappears), so everything works correctly in terms of behavior and appearance.

Thanks to this, when using only the search icon, I can style it using visualstate just like other taskbar buttons.
When the search pill is used, I only adjust properties like Padding and CornerRadius.
Likewise, with the search box, I also just tweak Padding and CornerRadius, since it refers to a different visualstate than the icon-only version.
This way, I can achieve the ideal look while avoiding crashes.

![explorer_1i2xNdZZb7](https://github.com/user-attachments/assets/ecadff6d-b078-4c08-83ed-07268b0ad0ed)
![KPZMWA8wT0](https://github.com/user-attachments/assets/53303770-c16c-4051-97e3-dca9a6f171fd)
![firefox_Vwfvp6wbui](https://github.com/user-attachments/assets/5dfecbb7-bda5-48fe-8d9f-e2f248d09dc5)

However, I noticed that when the XAML contains Taskbar.SearchBoxLaunchListButton, visualstate is ignored regardless of the current search button style. So the appearance might look a bit different from my setup.
But I don’t think that’s a big issue.

Thank you very much for the solution. I’ve updated the theme, and I’m happy if my input was helpful in any way.

m417z commented:
> However, I noticed that when the XAML contains Taskbar.SearchBoxLaunchListButton, visualstate is ignored regardless of the current search button style.

The visual states are ignored for `Taskbar.SearchBoxLaunchListButton > Taskbar.TaskListButtonPanel`, which I believe is only used for the "Icon and label" search button configuration. Both the search icon and the search box use `Taskbar.SearchBoxButton > Taskbar.TaskListButtonPanel`.

Giyuu12 commented:
> With the icon, I don't see a blue background, and the hover color is different than for other buttons:

thank you. I set up a test environment and found that the previous XAML structure was still being used, so I made adjustments while checking the behavior myself.

m417z commented:
Is the theme ready to be merged? If it is, please add it to the main readme in [the themes section](https://github.com/ramensoftware/windows-11-taskbar-styling-guide#themes) and come up with a screenshot (`screenshot.png`), it can be just the light/dark variant, or perhaps some image which shows both variants.
